submarine-security/spark-security/spark-2/src/main/scala/org/apache/submarine/spark/security/parser/SubmarineSqlParserCompatible.scala

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.submarine.spark.security.parser
+
+import org.apache.spark.sql.catalyst.parser.ParserInterface
+
+class SubmarineSqlParserCompatible(override val delegate: ParserInterface) extends SubmarineSqlParser(delegate: ParserInterface) {
```

Comment:
what is this for?

Reply:
in spark-3.x `ParserInterface` interface added 2 method : `parseMultipartIdentifier` and  `parseRawDataType`, this change for compatible spark2.x and spark3.x

Reply:
`SubmarineSqlParser` is a public API, it's not a good approach here

Reply:
if  `SubmarineSqlParser ` is public , has someone used it, we can Compatible `SubmarineSqlParser` class, not subclass extends it ?
move class ` SubmarineSqlParser` to spark2.x and spark3.x, has different implementations : 
spark3.x add method : `parseMultipartIdentifier` and `parseRawDataType`;
spark2.x same with `org.apache.submarine.spark.security.parser.SubmarineSqlParser`

this approach better, any have other idea?